File: backend/slotify/spring/src/main/java/com/slotify/backend/spring/reserve/dtos/TimeIntervalDTO.java
package com.slotify.backend.spring.reserve.dtos;

import java.time.LocalTime;

public record TimeIntervalDTO(LocalTime startTime, LocalTime endTime) {

    public TimeIntervalDTO {
        if (startTime != null && endTime != null && startTime.isAfter(endTime)) {
            throw new IllegalArgumentException("El startTime no puede ser posterior al endTime");
        }
    }

    /**
     * Verifica si una hora específica cae dentro de este intervalo.
     */
    public boolean contains(LocalTime time) {
        if (time == null) return false;
        return !time.isBefore(startTime) && !time.isAfter(endTime);
    }
}

File: frontend/app/src/main/java/com/raj/slotify/dtos/reserves/TimeIntervalDTO.kt
package com.raj.slotify.dtos.reserves

import com.google.gson.annotations.SerializedName
import java.time.LocalTime

data class TimeIntervalDTO (
    @SerializedName("startTime")
    val startTime: LocalTime,

    @SerializedName("endTime")
    val endTime: LocalTime
)